product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/vfs/VFSClient.java

```diff
@@ -0,0 +1,229 @@
+     * @param password password corresponds to the username
+     * @param port ftp port
+     */
+    public VFSClient(String hostname, String username, String password, int port) {
+        this.hostname = hostname;
+        this.userName = username;
+        this.password = password;
+        this.port = port;
+        this.ftpClient = new FTPClient();
+    }
+
+    /**
+     * Function to connect to an FTP Server.
+     *
+     * @throws IOException if connecting to FTP server fails
+     */
+    public void connectToFtp() throws IOException {
+        try {
+            ftpClient.connect(hostname, port);
+            ftpClient.login(userName, password);
+            ftpClient.enterLocalPassiveMode();
+        } catch (IOException ex) {
+            throw new IOException("Error occurred while connecting to FTP server", ex);
+        }
+    }
+
+    /**
+     * Function to upload a file to an FTP Server.
+     *
+     * @param sourceFilePath source file path
+     * @param targetFilePath target path of the remote file
+     * @throws IOException if file upload fails
+     */
+    public void ftpFileUpload(String sourceFilePath, String targetFilePath) throws IOException {
+        ftpClient.setFileType(FTP.BINARY_FILE_TYPE);
+        File localFile = new File(sourceFilePath);
+
+        try (InputStream inputStream = new FileInputStream(localFile)) {
+            log.info("Start uploading file " + sourceFilePath + " to " + targetFilePath);
+
+            try {
+                if (ftpClient.storeFile(targetFilePath, inputStream)) {
+                    log.info("File has been uploaded to " + targetFilePath + " successfully.");
+                }
+            } catch (IOException ex) {
+                throw new IOException("Error occurred while uploading file from " + sourceFilePath + " to " +
+                                      targetFilePath, ex);
+            }
+        } catch (FileNotFoundException ex) {
+            throw new IOException("File Not Found in " + sourceFilePath, ex);
+        }
+    }
+
+    /**
+     * Function to download a file from an FTP Server.
+     *
+     * @param sourceFilePath source file path
+     * @param targetFilePath target path
+     * @throws IOException if file download fails
+     */
+    public void ftpFileDownload(String sourceFilePath, File targetFilePath) throws IOException{
+        ftpClient.setFileType(FTP.BINARY_FILE_TYPE);
+
+        try (OutputStream outputStream = new BufferedOutputStream(new FileOutputStream(targetFilePath))) {
+            log.info("Start downloading file " + sourceFilePath + " to " + targetFilePath);
+
+            try {
+                if (ftpClient.retrieveFile(sourceFilePath, outputStream)) {
+                    log.info("File has been uploaded to " + targetFilePath + " successfully.");
+                }
+            } catch (IOException ex) {
+                throw new IOException("Error occurred while downloading file from " + sourceFilePath + " to " +
```

Comment:
Shall we break the line like below (+ should be at the begining of the line)

"Error occurred while downloading file from " + sourceFilePath 
 + " to + ...

Reply:
Fixed by PR [https://github.com/wso2/product-ei/pull/3466](https://github.com/wso2/product-ei/pull/3466)